### ISSUE TITLE
Reasoning for #10 is wrong.

### ISSUE THREAD

**Issue by g-w1** [ratfactor/ziglings#1]
https://github.com/ratfactor/ziglings/blob/main/exercises/10_if2.zig#L12

Here you say the reason that it fails is that the compiler is not smart enough, but this is not true.
The error is 
```
./main.zig:4:1: error: variable of type 'comptime_int' must be const or comptime
var i = if (true) 1 else 2;
^
./main.zig:2:29: note: referenced here
pub fn main() anyerror!void {
                            ^
```
The reason is that you cannot have a comptime_int variable in a non comptime scope. The compiler is smart enough to do peer type resolution.
For example, this works:
```zig
var b: bool = true;
var x = if (b) null else @as(u32, 5);
if (@TypeOf(x) != ?u32) @panic("OOF");
```
This example should be fixed and an explanation of peer type resolution should probably be made.

**Comment by ratfactor** [comment]
Thanks! This is exactly the kind of feedback I'm looking for. I'll correct it tonight.

**Comment by ratfactor** [comment]
@g-w1 I wrote up a whole new comment before I realized the entire issue should be ignored at this stage - I've removed the fallacious explanation _and_ the warning.  Until now, we've been doing the right thing (giving types to all integer vars).

Peer resolution is working correctly in this expression, as you mentioned.  I'll probably need to add at least one exercise later that will bring the issue up properly in its own context!

Thanks again for spotting this. 👍
